cmake/Utils.cmake

```diff
@@ -2,17 +2,9 @@
 # Add MSVC RunTime Flag
 function(add_msvc_runtime_flag lib)
   if(${ONNX_USE_MSVC_STATIC_RUNTIME})
-    if(${CMAKE_BUILD_TYPE} MATCHES "Debug")
-      target_compile_options(${lib} PRIVATE /MTd)
-    else()
-      target_compile_options(${lib} PRIVATE /MT)
-    endif()
+    target_compile_options(${lib} PRIVATE $<$<NOT:$<CONFIG:Debug>>:/MT> $<$<CONFIG:Debug>:/MTd>)
```

Comment:
What's the benefit of changing to use generator expression for this? IMO this new code looks more cryptic than the old version.

Reply:
https://stackoverflow.com/questions/34490294/what-does-configdebugrelease-mean-in-cmake